xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -195,7 +195,11 @@ CGDirectDisplayID GetDisplayID(int screen_index)
 
   // Get the list of displays.
   CGGetActiveDisplayList(MAX_DISPLAYS, displayArray, &numDisplays);
-  return(displayArray[screen_index]);
+
+  if (screen_index >= 0 && screen_index <= numDisplays)
```

Comment:
Ah wait - the check needs to be screen_index < numDisplays...

Reply:
ups, you're right. Fixed.